src/rendering/renderers/shared/texture/RenderableGCSystem.ts

```diff
@@ -130,12 +131,14 @@ export class RenderableGCSystem implements System<RenderableGCSystemOptions>
 
     public addRenderable(renderable: Renderable, instructionSet: InstructionSet): void
     {
+        if (!this.enabled) return;
+
         renderable._lastUsed = this._now;
 
         if (renderable._lastInstructionTick === -1)
         {
-            // TODO manage index...
             this._managedRenderables.push(renderable);
+            renderable.once('destroyed', this._removeRenderable, this);
```

Comment:
one thought on this (which i guess is a little subjective!) -  do we need to add the event listener? As this will always get cleaned up in the end either way?

Reply:
For this issue this PR was trying to solve, yes. Memory was being eaten up faster than the custom GC was running. If the question is whether or not we should address the issue itself, I believe it would be confusing if calling `destroy()` on an object still kept a reference to the object around under the hood for 30 seconds. Conceptually, I would expect this to fully destroy the object and remove all references.